arrows/vxl/aligned_edge_detection.cxx

```diff
@@ -0,0 +1,320 @@
+// This file is part of KWIVER, and is distributed under the
+// OSI-approved BSD 3-Clause License. See top-level LICENSE file or
+// https://github.com/Kitware/kwiver/blob/master/LICENSE for details.
+
+#include "aligned_edge_detection.h"
+
+#include <arrows/vxl/image_container.h>
+
+#include <vil/algo/vil_gauss_filter.h>
+#include <vil/algo/vil_sobel_3x3.h>
+#include <vil/vil_convert.h>
+#include <vil/vil_image_view.h>
+#include <vil/vil_math.h>
+#include <vil/vil_plane.h>
+
+#include <limits>
+#include <type_traits>
+
+#include <cstdlib>
+
+namespace kwiver {
+
+namespace arrows {
+
+namespace vxl {
+
+// ----------------------------------------------------------------------------
+/// Private implementation class
+class aligned_edge_detection::priv
+{
+public:
+  priv( aligned_edge_detection* parent ) : p{ parent } {}
+
+  // Calculate potential edges
+  template < typename PixType >
+  void
+  calculate_aligned_edges( vil_image_view< PixType > const& input,
+                           vil_image_view< PixType >& output_i,
+                           vil_image_view< PixType >& output_j );
+  // Perform NMS on the input gradient images in horizontal and vertical
+  // directions only
+  template < typename OutputType, typename InputType >
+  void
+  nonmax_suppression( vil_image_view< InputType > const& grad_i,
+                      vil_image_view< InputType > const& grad_j,
+                      vil_image_view< OutputType >& output_i,
+                      vil_image_view< OutputType >& output_j );
+  // Compute axis-aligned edges and perform NMS on them
+  template < typename pix_t > vil_image_view< pix_t >
```

Comment:
The `template` spec should probably not be on the same line as the return type. I guess the return type should be on its own line to match the above, but on the same line as the rest of the prototype is also okay and still within length; feel free to pick whichever you prefer.

Reply:
(Ping?)